workers/api-server/src/http/wallet.rs

```diff
@@ -57,6 +58,34 @@ pub(super) fn get_current_timestamp() -> u64 {
     now.duration_since(UNIX_EPOCH).unwrap().as_millis() as u64
 }
 
+/// Find the wallet for the given id in the global state
+///
+/// Attempts to acquire the lock for an update on the wallet
+async fn find_wallet_for_update(
+    wallet_id: WalletIdentifier,
+    state: &RelayerState,
+) -> Result<Wallet, ApiServerError> {
+    // Find the wallet in global state and use its keys to authenticate the request
+    let wallet = state
+        .read_wallet_index()
+        .await
+        .get_wallet(&wallet_id)
+        .await
+        .ok_or_else(|| {
+            ApiServerError::HttpStatusCode(StatusCode::NOT_FOUND, ERR_WALLET_NOT_FOUND.to_string())
+        })?;
+
+    // Acquire the lock for the wallet
+    if wallet.is_locked() {
```

Comment:
looks like we just check if the wallet is locked here, do we also want to `try_lock_wallet`?

Reply:
no, we want to centralize the actual locking mechanism